code/modules/antagonists/roundstart_special/undercover/undercover.dm

```diff
@@ -0,0 +1,130 @@
+////////////////////////////////
+////// Special Role Datum///////
+////////////////////////////////
+
+/datum/special_role/undercover
+	probability = 65			//The probability of any spawning at all
+	proportion = 0.05			//The prbability per person of rolling it (5% is (5 in 100) (1 in 20))
+	max_amount = 4				//The maximum amount
+	role_name = "Undercover Agent"
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Head of Personnel", "Chief Medical Officer", "Chief Engineer", "Research Director", "Captain", "Brig Physician", "Clown")
+	attached_antag_datum = /datum/antagonist/special/undercover
+
+////////////////////////////////
+//////  Antagonist Datum ///////
+////////////////////////////////
+
+/datum/antagonist/special/undercover
+	name = "Ex-security agent"
+	roundend_category = "Special Roles"
+	antag_moodlet = /datum/mood_event/determined
+
+/datum/antagonist/special/undercover/greet()
```

Comment:
I feel like these messages should be stored in their notes too. Does anybody actually read their notes?

Reply:
I read my notes

Reply:
You mean the great text saying you aren't an antag?

Reply:
The notes verb

Reply:
No I mean what text should go into the notes verb, I presume the you aren't an antag message